pkg/gateway/model_build_targetgroup.go

```diff
@@ -341,6 +355,15 @@ func (t *latticeServiceModelBuildTask) buildTargetGroupSpec(ctx context.Context,
 				return latticemodel.TargetGroupSpec{}, err
 			}
 		}
+
+		var err error
+
+		ipAddressType, err = buildTargetGroupIpAddressType(ctx, svc)
+
+		// Ignore error for creation request
```

Comment:
not clear why should ignore error

Reply:
also looking into buildTargetGroupIpAddressType you return dual stack error, which seems to be hard stop, confusing

Reply:
For deletion requests, we still want to build the model just like the code above

```go
		svc := &corev1.Service{}
		if err := t.Client.Get(ctx, serviceNamespaceName, svc); err != nil {
			glog.V(6).Infof("Error finding backend service %v error :%v \n", serviceNamespaceName, err)
			if !isDeleted {
				//Return error for creation request only,
				//For k8sService deletion request, we should go ahead to build TargetGroupSpec model
				return latticemodel.TargetGroupSpec{}, err
			}
		}
```

Reply:
Sorry I dont get it. buildTargetGroupIpAddressType returns single error - dual stack exception which seems critical error. This is the only place I see you use buildTargetGroupIpAddressType. So the error does not make any difference.
So you either need to handle error, or dont return it.

Reply:
`// Ignore error for creation request`

Your comment should be : `// Ignore error for deletion request` ?

Reply:
I think Scott's current code could work. Or change to this still work: 
```ipAddressType, _ = buildTargetGroupIpAddressType(ctx, svc)```

for the targetGroup deletion request, the `svc` is empty and buildTargetGroupIpAddressType(ctx, svc) return `"", err`, and in the `target_group_manager.go` it is able to handle the `targetGroup.Spec.Config.IpAddressType == ""`

For dual stack validation, `areBackendRefsIpFamiliesValid()` could handle it and  retrun early at  httproute_controller.go before  doing the`buildAndDeployModel()`

Currently the `buildTargetGroupSpec()` have a little bit high cyclomatic complexity and hard to read, we could refactor it and make it more clear to show how it build targetGroupSpec following 4 scenarios:
1.  ServiceImport creation
2. ServiceImport deletion
3. k8sService creation
4. k8sService deletion

Reply:
lets keep it simple, please remove this error from buildTargetGroupIpAddressType since you dont use it

Reply:
We still use the error in the case of Creation requests